libraries/AP_RSSI/AP_RSSI.cpp

```diff
@@ -155,11 +155,48 @@ float AP_RSSI::read_receiver_rssi()
 }
 
 // Only valid for RECEIVER type RSSI selections. Returns -1 if protocol does not provide link quality report.
-float AP_RSSI::read_receiver_link_quality()
+int16_t AP_RSSI::read_receiver_link_quality()
 {
     if (RssiType(rssi_type.get()) == RssiType::RECEIVER) {
         return RC_Channels::get_receiver_link_quality();
     }
+    // should never get to here
+    return -1;
+}
+
+int8_t AP_RSSI::read_receiver_rfmode()
```

Comment:
Generally we make this kind of function return bool to indicate success or failure and take a reference to get the result

Reply:
I was following how it was done with link quality, and also most of the other function is that calls is expecting a return of value or -1 if it shouldn't call it.